runtime/lua/vim/treesitter.lua

```diff
@@ -462,7 +462,7 @@ end
 ---
 ---@param bufnr (integer|nil) Buffer to stop highlighting (default: current buffer)
 function M.stop(bufnr)
-  bufnr = bufnr or api.nvim_get_current_buf()
+  bufnr = bufnr and bufnr ~= 0 and bufnr or api.nvim_get_current_buf()
```

Comment:
Avoid these precedence quizzes, use parens to make it obvious.
```suggestion
  bufnr = (bufnr and bufnr ~= 0) and bufnr or api.nvim_get_current_buf()
```